api/src/certification/configuration/domain/models/CertificationFrameworksChallenge.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * @param {Object} params
+ * @param {ComplementaryCertificationKey} params.complementaryCertificationKey
+ * @param {number} params.alpha
+ * @param {number} params.delta
+ * @param {string} params.challengeId
+ * @param {Date} params.createdAt
+ */
+
+export class CertificationFrameworksChallenge {
+  constructor({ createdAt, challengeId, alpha, delta, complementaryCertificationKey }) {
```

Comment:
Il faut mettre la JSDoc sur le constructeur pour activer l'Intellisense des IDE.

Concernant les parametres optionnels ou non, pour indiquer aux IDE que ce ne sont pas des [invariants](https://en.wikipedia.org/wiki/Class_invariant) il faut metre des crochets.

```suggestion


export class CertificationFrameworksChallenge {

/**
 * @param {Object} params
 * @param {ComplementaryCertificationKey} params.complementaryCertificationKey
 * @param {number} [params.alpha]
 * @param {number} [params.delta]
 * @param {string} params.challengeId
 * @param {Date} params.createdAt
 */
  constructor({ createdAt, challengeId, alpha, delta, complementaryCertificationKey }) {
```

Le resultats dans une IDE

![image](https://github.com/user-attachments/assets/12ba8537-5bbb-4c25-a30a-3e56f66ffbd7)